src/mongocxx/test/client_side_encryption.cpp

```diff
@@ -1999,4 +2000,427 @@ TEST_CASE("KMS TLS Options Tests", "[client_side_encryption][!mayfail]") {
+    }
+
+    // Drop and create the collection keyvault.datakeys.
+    {
+        write_concern wc_majority;
+        wc_majority.acknowledge_level(write_concern::level::k_majority);
+
+        auto coll = client["keyvault"]["datakeys"];
+        coll.drop(wc_majority);
+
+        const auto empty_doc = make_document();
+        client["keyvault"].create_collection("datakeys", empty_doc.view(), wc_majority);
+    }
+
+    // Insert key1Document in keyvault.datakeys with majority write concern.
+    {
+        write_concern wc_majority;
+        wc_majority.acknowledge_level(write_concern::level::k_majority);
+
+        options::insert insert_opts;
+        insert_opts.write_concern(std::move(wc_majority));
+
+        client["keyvault"]["datakeys"].insert_one(key1_document, insert_opts);
+    }
+
+    // Create a MongoClient named keyVaultClient.
+    // Create a ClientEncryption object named clientEncryption with these
+    // options:
+    //
+    // ClientEncryptionOpts {
+    //    keyVaultClient: <keyVaultClient>;
+    //    keyVaultNamespace: "keyvault.datakeys";
+    //    kmsProviders: { "local": { "key": <base64 decoding of LOCAL_MASTERKEY> } }
+    // }
+    options::client_encryption ce_opts;
+    ce_opts.key_vault_client(key_vault_client);
+    ce_opts.key_vault_namespace({"keyvault", "datakeys"});
+    ce_opts.kms_providers(_make_kms_doc(false));
+    client_encryption client_encryption(std::move(ce_opts));
+
+    // Create a MongoClient named encryptedClient with these AutoEncryptionOpts:
+    //
+    // AutoEncryptionOpts {
+    //    keyVaultNamespace: "keyvault.datakeys";
+    //    kmsProviders: { "local": { "key": <base64 decoding of LOCAL_MASTERKEY> } }
+    //    bypassQueryAnalysis: true
+    // }
+    options::auto_encryption auto_encrypt_opts{};
+    auto_encrypt_opts.key_vault_namespace({"keyvault", "datakeys"});
+    auto_encrypt_opts.kms_providers(_make_kms_doc(false));
+    auto_encrypt_opts.bypass_query_analysis(true);
+    options::client encrypted_client_opts;
+    encrypted_client_opts.auto_encryption_opts(std::move(auto_encrypt_opts));
+    class client encrypted_client {
+        uri{}, test_util::add_test_server_api(encrypted_client_opts)
+    };
+
+    return std::make_tuple(std::move(client_encryption), std::move(encrypted_client));
+}
+
+// https://github.com/mongodb/specifications/blob/master/source/client-side-encryption/tests/README.rst
+TEST_CASE("Explicit Encryption", "[client_side_encryption]") {
+    instance::current();
+
+    class client conn {
+        mongocxx::uri{}, test_util::add_test_server_api()
+    };
+
+    if (!mongocxx::test_util::should_run_client_side_encryption_test()) {
+        return;
+    }
+
+    if (!test_util::newer_than(conn, "6.0")) {
+        std::cerr << "The Explicit Encryption tests require MongoDB server 6.0+." << std::endl;
+        return;
+    }
+
+    if (!test_util::is_replica_set(conn)) {
```

Comment:
Small issue, I'm seeing a lot of test failures with `if (test_util::get_topology(conn) != "single") {`. The tests are failing with the following error:
```
[2023/01/09 20:23:39.604] -------------------------------------------------------------------------------
[2023/01/09 20:23:39.604] Explicit Encryption
[2023/01/09 20:23:39.604] -------------------------------------------------------------------------------
[2023/01/09 20:23:39.604] /data/mci/2ba7dc145cf5a4dfa7622cffe867f842/mongo-cxx-driver/src/mongocxx/test/client_side_encryption.cpp:2086
[2023/01/09 20:23:39.604] ...............................................................................
[2023/01/09 20:23:39.611] /data/mci/2ba7dc145cf5a4dfa7622cffe867f842/mongo-cxx-driver/src/mongocxx/test/client_side_encryption.cpp:2086: FAILED:
[2023/01/09 20:23:39.611]   {Unknown expression after the reported line}
[2023/01/09 20:23:39.611] due to unexpected exception with message:
[2023/01/09 20:23:39.611]   Encrypted collections are not supported on standalone: generic server error
[2023/01/09 20:23:40.041] -------------------------------------------------------------------------------
```
If I switch it back to `if (!test_util::is_replica_set(conn)) {` then tests pass. Do you have an idea of why this could be?

Reply:
Haha, no worries! Thanks!